examples/pio_blink/main.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include "board.h"
+#include "periph/pio.h"
+#include "shell.h"
+
+pio_program_t pio_blink_export_program(void);
+
+int pio_blink_init(pio_program_t *blink,
+                   pio_t pio, pio_sm_t sm, const pio_program_t *pro,
+                   gpio_t pin);
+
+static const shell_command_t shell_commands[] = {
+    {NULL, NULL, NULL}
+};
+
+int main(void)
+{
+    pio_program_t blink = pio_blink_export_program();
+    pio_t pio;
+    pio_sm_t sm;
+    if (pio_alloc_program_sm_lock_any(&pio, &sm, &blink)) {
+        puts("No PIO resources available.");
+        return 1;
+    }
+    if (pio_blink_init(&blink, pio, sm, &blink, LED0_PIN)) {
+        puts("PIO blink initialization failed");
+        return 1;
+    }
+    pio_sm_start(pio, sm);
+
+    static char line_buf[SHELL_DEFAULT_BUFSIZE];
+    shell_run_forever(shell_commands, line_buf, SHELL_DEFAULT_BUFSIZE);
```

Comment:
a shell without a usecase, is a bit complex for a sleep forever ;-)

Reply:
Maybe we could keep this for future commands - for example ones that change clock divider (blink frequency) or used pin?

Reply:
Good idea. However I removed it for now.